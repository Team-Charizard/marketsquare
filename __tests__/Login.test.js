import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import Login from '../client/components/Login';

describe('Login unit tests', () => {
  let wrapper;

  beforeAll(() => {
    wrapper = shallow(<Login />);
  });

  it('snapshot test', () => {
    const tree = renderer.create(<Login />).toJSON();
    expect(tree).toMatchSnapshot();
    expect(wrapper).toMatchSnapshot();
  });

  it('renders a div with the class name login-container', () => {
    console.log(wrapper.debug());
    expect(wrapper.type()).toEqual('div');
    expect(wrapper.find(`.login-container`).length).toEqual(1);
  });

  it('renders market square login', () => {
    const h2 = wrapper.find('h2');
    expect(h2.text()).toEqual('MarketSquare Login');
  });

  it('renders a login form', () => {
    const form = wrapper.find('form');
    expect(form.length).toEqual(1);
  });
});
