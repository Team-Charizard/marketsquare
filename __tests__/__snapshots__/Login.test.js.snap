// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login unit tests snapshot test 1`] = `
<div
  className="login-container"
>
  <div
    className="login-div"
  >
    <h2>
      MarketSquare Login
    </h2>
    <form
      className="login-form"
      onSubmit={[Function]}
    >
      <input
        name="username"
        onChange={[Function]}
        placeholder="enter username"
        type="text"
      />
      <input
        name="password"
        onChange={[Function]}
        placeholder="password"
        type="text"
      />
      <button
        id="login-button"
        type="submit"
      >
        Login
      </button>
    </form>
  </div>
</div>
`;

exports[`Login unit tests snapshot test 2`] = `
<div
  className="login-container"
>
  <div
    className="login-div"
  >
    <h2>
      MarketSquare Login
    </h2>
    <form
      className="login-form"
      onSubmit={[Function]}
    >
      <input
        name="username"
        onChange={[Function]}
        placeholder="enter username"
        type="text"
      />
      <input
        name="password"
        onChange={[Function]}
        placeholder="password"
        type="text"
      />
      <button
        id="login-button"
        type="submit"
      >
        Login
      </button>
    </form>
  </div>
</div>
`;
